workflow/templateresolution/context.go

```diff
@@ -135,6 +162,12 @@ func (ctx *Context) GetTemplateFromRef(tmplRef *wfv1.TemplateRef) (*wfv1.Templat
 		}
 		return nil, err
 	}
+	if !tmplRef.ClusterScope && wftmpl != nil {
```

Comment:
Here we don't update ClusterWorkflowTemplates (`!tmplRef.ClusterScope`). They should be updated too. It means cwftmplClient which you've added doesn't do any work ever.

Reply:
How about adding the feature of updating ClusterWorkflowTemplates' status in a separate PR?
It will be easier for me to manage

Reply:
I'd want the two merged at the same time, so they'd need to be stacked PRs which is hard to manage too. Get it right for non-cluster and the clusterwftemplates should be pretty simple to add at the end.

Reply:
OK, I got it.